src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,214 @@
+            children: items.length !== 0 ? items.map(item => ({
+                id: item.name,
+                name: item.displayName,
+                checkProps: { checked: checkAll || currentFilterChipValues?.includes(item.name) }
+            })) : undefined
+        };
+    });
+
+    return init;
+};
+
+export const EventLogTreeFilter: React.FunctionComponent<EventLogTreeFilterProps> = (props) => {
+    const { groups, placeholder, filters, updateFilters } = props;
+
+    const initialize = React.useMemo(() => initTreeNodeById(groups, filters), [ groups, filters ]);
+
+    const [ treeNodeById, setTreeNodeById ] = React.useState<TreeNodeDict>(initialize);
+    const [ isToggled, setIsToggled ] = React.useState(false);
+
+    const treeDataArray = React.useMemo(() => !!treeNodeById ? Object.values(treeNodeById) : [], [ treeNodeById ]);
+
+    React.useEffect(() => {
+        if (groups.length !== 0) {
+            setTreeNodeById(produce((prev) => {
+                // In case Bundles request is pending during Tree View render
+                if (Object.keys(prev).length === 0) {
+                    return initialize;
+                }
+                else if (filters.length === 0) {
+                    return initialize;
+                }
+                else {
+                    filters.forEach(activeFilter => {
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        }
+                        else if (childChecked(treeNode)) {
+                            treeNode.checkProps.checked = null;
+                        }
+                        else {
+                            treeNode.checkProps.checked = false;
+                        }
+                    });
+                }
+            }));
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ groups, filters ]);
+
+    const flattenTree = React.useMemo(() => {
+        const flatTreeDataArray = produce(treeDataArray, (prev) => {
+            prev.forEach(treeNode => {
+                if (!!treeNode.children) {
+                    prev.push(...treeNode.children);
+                }
+            });
+        });
+
+        return flatTreeDataArray;
+    }, [ treeDataArray ]);
+
+    React.useEffect(() => {
+        const groupNames = groups.map(group => group.name);
+        const activeParentFilters = flattenTree.filter(treeNode => {
+            const isActive = treeNode.checkProps.checked;
+            return (!!isActive || isActive === null) && groupNames.includes(treeNode.id);
```

Comment:
```suggestion
            return (isActive || isActive === null) && groupNames.includes(treeNode.id);
```

Also, why `isActive` is null?

Reply:
For the TreeView patternfly component, null signifies a "half checked state" which means if any of the children are checked it displays a 'minus' sign to signify that it is being applied but not all of the children filters. 

Quick image of what I'm talking about.

![image](https://user-images.githubusercontent.com/80718590/162004736-76c3c55c-7b6d-4cfa-bc6e-c996e85eb8fc.png)